src/components/Layout/PageHeader.vue

```diff
@@ -146,9 +146,23 @@ export default {
         async leaveGroup() {
             await this.patch([this.user.username, { workshops: [], numberInGroup: null }]);
         },
+
+        logoutStudentAfterBrowserIsClosed(event) {
+            if (this.user.permissions.includes('student')) {
+                this.logoutUser()
+            }
+        },
     },
+
+    onIdle() {
```

Comment:
To avoid repetition, id use the above function here. (Since it will no longer be used only on browser closing, you can shorten its name.)

```
if (this.user) {
    this.logoutStudent();
},
```

Also, add missing semicolons.